control/controller_test.go

```diff
@@ -189,6 +222,14 @@ func TestController_BlockedExecutor(t *testing.T) {
 	ctrl := New(Config{})
 	ctrl.executor = executor
 
+	cctx, ccancel := context.WithTimeout(context.Background(), time.Second)
+	defer func() {
+		if err := ctrl.Shutdown(cctx); err != nil {
```

Comment:
Does this work because cctx is the top level context?

Reply:
No, it is just used to prevent a deadlock from stopping the cleanup. So this context isn't part of the test.